internal/server/deviceflow.go

```diff
@@ -50,6 +49,10 @@ retry:
 	if rctx.Authenticated.Organization != nil {
 		host = rctx.Authenticated.Organization.Domain
 	}
+	if host == "" {
+		// Default to the request hostname when in single tenant mode
+		host = fmt.Sprintf("https://%v", rctx.Request.URL.Host)
+	}
```

Comment:
This is the fix for the url being wrong in single tenant mode.

TODO: write a test case to over this.

Reply:
I added the test for this in the same commit with this fix. There was a bug here (the scheme was being added twice).